src/magic_spell_effect.cpp

```diff
@@ -178,6 +178,25 @@ static bool in_spell_aoe( const tripoint &start, const tripoint &end, const int
     return true;
 }
 
+void spell_effect::teleport_to( const spell &sp, Creature &caster, const tripoint &target )
```

Comment:
So i'm looking at this spell effect and it's actually _too_ similar to the other spell effect. I would like this spell effect merged into the old one. Here's how:
- make a new flag to change that `variance` actually is the distance from the target point instead of distance to the caster
- add the TARGET_RANDOM flag to the previous spells that were random teleport
- update the name of it to short_range_teleport or something similar, i don't want to leave "random" in there if it's only random with an added flag.